tasks/main.yml

```diff
@@ -115,8 +118,8 @@
   - name: "Configure /etc/hosts"
     lineinfile:
       dest: /etc/hosts
-      line:  "127.0.0.1    localhost {{ ansible_fqdn }} {{ ansible_nodename }} {{ ansible_hostname }}"
-      regexp: '^127\.0\.0\.1'
+      line:  "127.0.1.1    {{ ansible_fqdn }} {{ ansible_nodename }}"
```

Comment:
This change seemed good on the surface, but in testing on the develop branch, it yields the following results on an upgraded server:

```
Mar 10 20:48:59 domain.redacted.com RocketChat[1035]: /var/lib/rocket.chat/bundle/programs/server/node_modules/fibers/future.js:313
Mar 10 20:48:59 domain.redacted.com RocketChat[1035]:                                                 throw(ex);
Mar 10 20:48:59 domain.redacted.com RocketChat[1035]:                                                 ^
Mar 10 20:48:59 domain.redacted.com RocketChat[1035]: MongoError: not master and slaveOk=false
Mar 10 20:48:59 domain.redacted.com RocketChat[1035]:     at Function.MongoError.create (/var/lib/rocket.chat/bundle/programs/server/npm/node_modules/meteor/npm-mongo/node_modules/mongodb-core/lib
Mar 10 20:48:59 domain.redacted.com RocketChat[1035]:     at queryCallback (/var/lib/rocket.chat/bundle/programs/server/npm/node_modules/meteor/npm-mongo/node_modules/mongodb-core/lib/cursor.js:21
Mar 10 20:48:59 domain.redacted.com RocketChat[1035]:     at /var/lib/rocket.chat/bundle/programs/server/npm/node_modules/meteor/npm-mongo/node_modules/mongodb-core/lib/connection/pool.js:469:18
Mar 10 20:48:59 domain.redacted.com RocketChat[1035]:     at _combinedTickCallback (internal/process/next_tick.js:131:7)
Mar 10 20:48:59 domain.redacted.com RocketChat[1035]:     at process._tickCallback (internal/process/next_tick.js:180:9)
```

Changing back to a single 127.0.0.1 line fixes it. 

Ideas?

Reply:
On my installation I had to change it to have only ```127.0.0.1 localhost``` in /etc/hosts. I also use ```rocket_chat_mongodb_server: 127.0.0.1``` in my playbook. I got the same error with my mongoDB before I changed the task, because mongoDB does some hostname lookups on startup to magically determine it's hostname and in my case it got it wrong after updating the ansible role.
As I have written in my commit message, on Debian you should have only localhost for 127.0.0.1 in your /etc/hosts, otherwise ```hostname -f``` will return localhost instead of the FQDN. https://www.debian.org/doc/manuals/debian-reference/ch05.en.html#_the_hostname_resolution

A solution would be to have two tasks to modify /etc/hosts, one for Debian and one for all others.
Or if you don't like it, you can cherry pick and I keep my local version for my (crippled) installation.

Reply:
Hey i came up with a different solution for this here: https://github.com/RocketChat/Rocket.Chat.Ansible/commit/444988c74931a54967a5b1c292c6ebdb735acc2e

It's still utilizing `127.0.1.1` for `{{ ansible_nodename }}` and `{{ ansible_hostname }}`, but now (and this is the critical part for it to work on upgradable hosts) it reconfigures the mongodb replset to point to the correct `{{ ansible_nodename }}`. I think this is as idempotent as I can possibly make it... can you test? (it should only run on `rocket_chat_include_mongodb == True` plays, so I don't see any fear of messing anything up? (I hope)

There are a **ton** of other debian-based and improvement changes on that `v3.1.0_updates` branch that primarily step from your PR here as well, so bear in mind that if you do choose to test it.

Reply:
```v3.1.0_updates``` looks good! I ran an upgrade of my rocket.chat installation on Debian9 from 0.62.1 to 0.62.2 (latest) with this branch and it worked well.
I have two little comments:
- The line in /etc/hosts should be ```127.0.1.1    {{ ansible_fqdn }} {{ ansible_nodename }}```. Now I got two times the short hostname in this line. ```ansible_nodename```and ```ansible_hostname``` should always be identical. 
- I know nearly nothing about mongodb, but shouldn't we always use ```rocket_chat_mongodb_server``` as member of the replset? In the task "Ensure the MongoDB replSets have been initiated" ```rocket_chat_mongodb_server``` is used, but in the task "Reset replSet config when /etc/hosts changes" you use ```ansible_nodename```.  ```rocket_chat_mongodb_server``` is also used in ```rocketchat.service.js2``` and ```mongod.conf.js2```. IMHO it should be consistent and in all places ```rocket_chat_mongodb_server``` should be used.

What do you think about it?

Reply:
Some quick feedback on your two points: 

1. The way those three vars works is: 
    * `ansible_nodename` : The full name as reported by the system
    * `ansible_hostname` : The first subdomain of `ansible_nodename` or equivalent
    * `ansible_fqdn`: This is where it gets tricky, this is like `hostname -f` and it calls `gethostbyname(3)` which is network aware. People unaware of this that have a name like `test.computer` will end up with the name: `auth1.junet.se`, why? because it's the PTR for `146.66.237.143` which in turn is what `test.computer` resolves to if it is not caught locally (point is this is something that's easy to mess up these days with all the gTLDs) ** It is not inherently bad to use this in ansible if it's placed in /etc/hosts _before_ trying to resolve it!* So, I don't think I agree with using it here. I feel that `ansible_nodename` should be the responsibility of the administrator to set properly. Still open to interpretation though... 

2. I'll have to check this over again myself. My thoughts are that `mongodb.yml` is to be used only when `rocket_chat_include_mongodb == False` and if they want a complicated replSet setup then that's outside of the scope of this role. Hence the configuration need not matter in terms of `rocket_chat_mongodb_server` because if this is set to something custom, it implies (or at least should) that they're going off-role to deal with MongoDB anyway. The reason I wanted to use `{{ ansible_nodename }}` there was because mongodb's documentation says to use the hostname and not localhost, as far as I can tell. Bottom line is, mongodb.yml is meant for a single-server single-mongodb deployment... Why replSets then? -- Well I don't remember exactly _why_, but there's some component with meteor that _requires_ it's initialized, I think something with logging. Thus, we have to go through the motions and enable it. At any rate, we need to update the `rs.config()` from now on when touching `/etc/hosts` because otherwise we risk mongodb starting up as REMOVED and then rocketchat doesn't want to run at all and it's quite confusing to figure out where it went wrong exactly. So, I'll check into the total usage of those vars and standardize it how I see fit.

Thank you for the feedback and collaboration, definitely appreciated!

Reply:
You are welcome!
I looked into the ansible code, ```ansible_nodename``` is the result of python's ```platform.node()``` and python docs say: "platform.node()
Returns the computer’s network name (may not be fully qualified!). An empty string is returned if the value cannot be determined."
And ```platform.node()``` uses just ```socket.gethostname()``` to determine the nodename. So it depends heavily on the actual system config, if we get a FQDN or a short hostname for ```ansible_nodename```. 

Different Linux distributions have different defaults for their network config...

I assume mongoDB also does ```socket.gethostname()``` to get it's nodename, so it should be the right thing to use ```ansible_nodename``` in the replset config.